### ISSUE TITLE
Laout/ExtraSpacing vs AllowForAlignment: more careful consideration of "adjacent" concept?

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#15554]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Laout/ExtraSpacing

### Is your feature request related to a problem?

Not sure if it should be considered a "bug" (so posting it as a Feature request rather), but there is a behavior of Layout/ExtraSpacing + AllowForAlignment which I consider pretty surprising.

A problematic fragment (ActiveAdmin API):

```ruby
  sidebar 'User Details', only: :show do
    attributes_table_for user_invoice.user do
      row(:name) { |u| u }
      row :email
      row :roles
      row :phone
      row :time_zone
      row(:deleted) { status_tag it.deleted } # <== this line is decided to be "aligned" to
    end
  end

  sidebar 'Author Details', only: :show do
    attributes_table_for user_invoice.author do
      row(:name) {  it }  # <== here is extra spacing (two spaces)
      row :email
      row :roles
      row :phone
      row :time_zone
      row(:deleted) { |a| status_tag a.deleted }
    end
  end
```

This code is considered having no offenses in the second line I marked. By debugging it (simply inserting a lot of `p` into the RuboCop codebase 🙈) I found out that the first line I've marked is considered as alignment anchor.

It seems that there are several problems with this judgement:

1. The lines are too distant from one another (even if they've been in the same block)
2. The lines are from different blocks (so the alignment of one to the other is doubtful)
3. The things that are decided to be aligned are of different "token types", so even if the lines are directly adjacent... I am not sure it is a probable alignment:
```ruby
      row(:deleted) { status_tag it.deleted }
      row(:name) {  it }
#                   ^ Ummm really?..
```

### Describe the solution you'd like

I am not sure which of the three problems above is the most glaring/easiest to unambiguously analyze and avoid. Ideally, all three, probably? But it seems to me that "different nesting blocks" is the most unambiguous. And "too far away" might be a settable threshold?.. While "different token types" might be not that obvious to automatic analysis, even if obvious to a human.

### Code examples

```ruby

```
